generator/transit_generator.cpp

```diff
@@ -198,8 +210,15 @@ void DeserializerFromJson::operator()(m2::PointD & p, char const * name)
   FromJSONObject(pointItem, "x", p.x);
   FromJSONObject(pointItem, "y", p.y);
 }
+DeserializerFromJson::DeserializerFromJson(json_struct_t * node,
+                                           shared_ptr<OsmIdToFeatureIdsMap> const & osmIdToFeatureIds)
+  : m_node(node), m_osmIdToFeatureIds(osmIdToFeatureIds)
```

Comment:
Why do we need shared pointer here? Why not to pass mapping by const ref?

Reply:
It's a possible solution and I thought about it. And yes. I'll be easier and more efficient.

On the other hand to implement it I need:
- to create an instance of `OsmIdToFeatureIdsMap` in `BuildTransit(..)` method
- to pass it by ref to `DeserializeFromJson(...)` and `SerializeObject(...)` method
- then to pass by ref to constructor of `DeserializerFromJson` and use it in it

I'm afraid of dangling references in the future. What do you think about that?

Reply:
What dangling references do you afraid? OsmIdToFeatureIds mapping should be alive for the whole generator_tool lifetime, what's the problem?

Reply:
I'm afraid of dangling ref which may appear in the future because of the code would be modified. Nevertheless I don't mind to rewrite it with const ref.

Reply:
Done